File: Data Structures & Algorithms/is-anagram/submission-3.java
class Solution {
    public boolean isAnagram(String s, String t) {
        if (s == null && t == null) {
            return true;
        }
        if (s == null || t == null || s.length() != t.length()) {
            return false;
        }
        char[] frequencies = new char[26];
        for (int i = 0; i < s.length(); i++) {
            frequencies[s.charAt(i) - 'a']++;
            frequencies[t.charAt(i) - 'a']--;
        }

        for (int frequency: frequencies) {
            if (frequency != 0) {
                return false;
            }
        }
        return true;
    }
}

File: Data Structures & Algorithms/top-k-elements-in-list/submission-3.kt
class Solution {
    fun topKFrequent(nums: IntArray, k: Int): IntArray {
        val frequencies = mutableMapOf<Int, Int>()
        nums.forEach {
            frequencies.merge(it, 1, Int::plus)
        }
        val groupedFrequents = Array<MutableList<Int>>(nums.size + 1) {mutableListOf<Int>()}
        frequencies.forEach {
            (value, frequency) -> groupedFrequents[frequency].add(value)
        }
        val topFrequent = IntArray(k)
        var n = k
        for (i in nums.size downTo 1) {
            groupedFrequents[i].forEach {
                topFrequent[n - 1] = it
                n--
                if (n == 0) {
                    return topFrequent
                }
            }
        }

        return topFrequent
    }
}
